### DISCUSSION TITLE
OpenFont Render and Flicker-free text?

### DISCUSSION THREAD

**Discussion by trailsurfer604** [Bodmer/TFT_eSPI#2267] [Q&A - Fonts]
I saw Bodmer's announcement about supporting OpenFontRender and got excited - easily scalable font support was just what I was hoping to find for my automotive sensor display project with 2.4" TFT.  The screen will display six digital measurements, which may be updated every 250 ms.  

My question is, is it possible to use OpenFontRender in a way to make the text flicker free as it's updated?

Bodmer previously recommended using [FlickerFreePrint](https://github.com/KrisKasprzak/FlickerFreePrint), but it seems to stand on its own and does not work with OpenFontRender ... or I haven't figured out how yet.

Failing that, what are my best options for flicker free text? 

I am currently using the method of overwriting text with tame text in BG color, and then printing new text with FG color, which works OK for white on black test, but for some reason does not work for black on white, leaves an alias behind.

**Comment by Bodmer** [comment]
Best option is to use Sprites. Example here:
https://github.com/takkaO/OpenFontRender/tree/master/examples/TFT_eSPI/Sprite_Noto_Font_Demo

To over-write an anti-aliased character directly on the screen both foreground and background of the text must be set to the same colour. In this case both white or both black.

**Reply by trailsurfer604** [reply]
> Best option is to use Sprites. Example here: https://github.com/takkaO/OpenFontRender/tree/master/examples/TFT_eSPI/Sprite_Noto_Font_Demo
> 
> To over-write an anti-aliased character directly on the screen both foreground and background of the text must be set to the same colour. In this case both white or both black.

I will check out sprites, thank you.

As for over-writing, I am using that method.  Black on Black works fine, but white on white does not for some reason.

**Reply by Bodmer** [reply]
Maybe there is a misunderstanding about setting both foreground and background colours to the same colour?

[Blank_demo_2267.zip](https://github.com/Bodmer/TFT_eSPI/files/10297358/Blank_demo_2267.zip)

**Comment by trailsurfer604** [comment]
I was doing the same, except I am using OpenFontRender rather than your built-in functions:

` render.setFontColor(bg_color);
  render.setCursor(BANNER_VAL1_X, BANNER_VAL_Y);
  render.printf("%.1lf", banner_val1);
  render.printf(BANNER_VAL1_UNIT);
        
 render.setFontColor(fg_color);
 render.setCursor(BANNER_VAL1_X, BANNER_VAL_Y);
 render.printf("%.1lf", t);
 render.printf(BANNER_VAL1_UNIT);
 banner_val1 = t;
`

This worked perfectly fine when bg_color was TFT_BLACK and FG_color was TFT_WHITE, but left aliases when I did it backwards.

**Reply by Bodmer** [reply]
The principles are the same, the key is the foreground background combinations. Here is the same demo for the OpenFontRender library, does it work with your setup?
[Blank_ofr_demo_2267.zip](https://github.com/Bodmer/TFT_eSPI/files/10297493/Blank_ofr_demo_2267.zip)

**Reply by trailsurfer604** [reply]
Thank you, I've been away for a few days, will try this.

**Reply by Adams5098** [reply]
Thanks
